File: python practice/20260614_Train_Shuffle.py
import sys

def solve():
    input_data = sys.stdin.read().split()
    if not input_data:
        return
    
    idx = 0
    n_elements = len(input_data)
    
    while idx < n_elements:
        n = int(input_data[idx])
        idx += 1
        
        A = input_data[idx : idx + n]
        idx += n
        
        B = input_data[idx : idx + n]
        idx += n
        
        left = 0
        right = n - 1
        possible = True
        
        for i in range(n - 1, -1, -1):
            if B[left] == A[i]:
                left += 1
            elif B[right] == A[i]:
                right -= 1
            else:
                possible = False
                break
        
        if possible:
            sys.stdout.write("Success\n")
        else:
            sys.stdout.write("Fail\n")

if __name__ == '__main__':
    solve()

File: C Language/20260614_Train_Shuffle.c
#include <stdio.h>
#include <string.h>

int main() {
    int n;
    char A[1005][50], B[1005][50];
    
    while (scanf("%d", &n) == 1) {
        for (int i = 0; i < n; ++i) scanf("%s", A[i]);
        for (int i = 0; i < n; ++i) scanf("%s", B[i]);

        int left = 0, right = n - 1;
        int possible = 1;
        
        for (int i = n - 1; i >= 0; --i) {
            if (strcmp(B[left], A[i]) == 0) {
                left++;
            } else if (strcmp(B[right], A[i]) == 0) {
                right--;
            } else {
                possible = 0;
                break;
            }
        }
        
        if (possible) {
            printf("Success\n");
        } else {
            printf("Fail\n");
        }
    }
    return 0;
}